### ISSUE TITLE
Incorrect import in app.js

### ISSUE THREAD

**Issue by kxxoling** [mxstbr/react-boilerplate#963]
I'm writing a web app on react-boilerplate. It works well in dev mode, and I fixed all the lint issues and tests. But when I trying to deploy it in production mode, I found it always catch this error in Browser (Chrome here) console:

    main.e413d39….js:22 Uncaught TypeError: b(...) is not a function

And this all that I can found with the issue. So how can I figure out what problem it is and how can I fixed it?

Source code of my app is here: https://github.com/kxxoling/me

```
node -v
v6.5.0

npm -v
3.10.3
```

**Comment by eavae** [comment]
I do some research about this issue, the problem is there have a wrong import in `app.js` file.

Change `import useScroll from 'react-router-scroll';` to `import {useScroll} from 'react-router-scroll';` will solve this problem.

Can any one create a PR for this?

**Comment by timgaines** [comment]
Here it is #956

**Comment by gihrig** [comment]
@mxstbr is 'on the road', I've been really busy with work, but If someone could merge #952 and #956 into their local project and confirm all works as expected, I'll go ahead and merge these.

**Comment by kxxoling** [comment]
I found something strange here: @eavae 's solution really fixed my issue in production, but breaks my code in dev mode, that's why I did not find this issue until the first time I tried to deploy it to GitHub pages.

**Comment by gihrig** [comment]
@kxxoling Are you seeing this with the latest `dev` branch?

I believe `dev` branch is ready to be released as the next react-boilerplate version 3.2.0, except for a minor lint issue [#948 (comment)](https://github.com/mxstbr/react-boilerplate/pull/948#issuecomment-246541141)

**Comment by kxxoling** [comment]
@gihrig How can I merge the latest dev branch on my project? I've been working with the project for a period.

**Comment by gihrig** [comment]
> How can I merge the latest dev branch on my project? I've been working with the project for a period.

@kxxoling Yeah, that's not going to be easy 😦 

I guess it depends a lot how much you have modified boilerplate components, e.g. webpack config, etc.

Another troubleshooting approach that probably makes more sense in your case is to roll back your project to earlier commits until the problem goes away, then compare the commit that created the problem. I find that a git GUI tool really helps with this. I use [GitUp](http://gitup.co), [GitKraken](https://www.gitkraken.com/) looks interesting also, both free.

If you decide to try and update the boilerplate, it will probably be easier to start with a new clone of react-boilerplate and copy your project into it than trying to update the boilerplate in your project. And there is the possibility that this will still not fix your problem.

See [#880 comment](https://github.com/mxstbr/react-boilerplate/issues/889#issuecomment-247073755)

**Comment by kxxoling** [comment]
@gihrig Have I update the whole base to fix this issue? I though this may could be updated just like a usual scaffold tool.

**Comment by gihrig** [comment]
I really don't know about fixing the issue, that was my guess as I have no idea what to make of the error you listed above.

But yes, react-boilerplate is not a library or really a 'scaffold tool'. It's is truly a boilerplate. Intended to be modified by the dev as the app progresses. That makes updating the project difficult, if not impractical.

You could try just updating dependencies, but that is known to cause several errors. You could learn more about those errors and the related corrections by searching the GitHub project for `is:pr update-deps `.

**Comment by kxxoling** [comment]
It's really a risk to rebase on a new boilerplate, I'd rather dig more about this issue first.

**Comment by uzarubin** [comment]
@kxxoling I ran into a cryptic error similar to this one: `h(...) is not a function`. Surprisingly, it was only failing when I tried to run the build process. I ended up resetting head and committing my changes one at a time until the error showed up. It turned out to be that I was using decorators for the `connect` function instead of just calling `connect(Container)`.

**Comment by mxstbr** [comment]
This is fixed on master. Sorry for the troubles!

**Comment by kxxoling** [comment]
@mxstbr I've cloned the newest version and created a new project from it. Then I tried to use the new staff to replace the old ones. I almost succeed except one error in dev mode which I think it's a bug of react-fa. After removing the react-fa part, it's OK in dev mode now.

Then I tried to build a new release with command ``npm run start:production`` which gives me no useful information as other npm logs.

Soon I find it's something wrong with the build process in ``npm run build``. Then I tried to use ``cross-env NODE_ENV=production webpack --config internals/webpack/webpack.prod.babel.js --color -p`` instead trying to get the useful traceback. What surprised me is it works this time! But I have no idea why.

But I intend not to dig into it now. I run ``npm run start:prod`` to see whether it's OK now. What I got is an empty page with error message ``main.20450d3….js:22 Uncaught Error: Cannot find module "containers/App/styles.css``. I find out the relative code lives in ``app/app.js``. I thought this maybe something wrong with webpack alias, so I used relative path instead which turns out no effect.

Anything that I can do to find out what's wrong with my code?

BTW, is there any better way to update react-boilerplate part in my project?
